### PR TITLE
:bug: (autocomplete) allow editing selected payee name

### PR DESCRIPTION
**Pull request by MatissJanis**
https://github.com/actualbudget/actual/issues/773

Before:

https://user-images.githubusercontent.com/886567/229928970-b243e7b9-0f28-41ed-8140-7cc4db26db8c.mov

After:

https://user-images.githubusercontent.com/886567/229929000-58fa8557-2c4d-48cd-87f4-21209e3aa5a4.mov

### CODE REVIEW

File: packages/desktop-client/src/components/autocomplete/NewAutocomplete.tsx [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
 import { NullComponent } from '../common';
 
 import styles from './autocomplete-styles';
 
-type OptionValue = string;
+type OptionValue = {
+  __isNew__?: boolean;
+  label: string;
+  value: true;
+};
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/NewAutocomplete.tsx:21]: Before: `OptionValue: string`

After: `OptionValue: ...`

I previously set the incorrect interface here.. oops. Fixing it in this PR.

File: packages/desktop-client/src/components/autocomplete/NewPayeeAutocomplete.js [hunk 2]
```diff
@@ -154,7 +150,6 @@ export default function PayeeAutocomplete({
                 }
               }
               onClick={() => {
-                setInputValue('');
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/NewPayeeAutocomplete.js:157]: This is a small downside to the current implementation. Theoretically we could duplicate all the `inputValue` manipulation logic in this file too, but.. that would be a lot of complex duplications I'd prefer not to do.

So I would recommend removing this line in favor of the greater good.

File: packages/desktop-client/src/components/autocomplete/NewAutocomplete.tsx [hunk 3]
```diff
@@ -1,27 +1,47 @@
...
+import type { CreatableProps } from 'react-select/creatable';
 import CreatableSelect from 'react-select/creatable';
 
 import { NullComponent } from '../common';
 
 import styles from './autocomplete-styles';
 
-type OptionValue = string;
+type OptionValue = {
+  __isNew__?: boolean;
```
Comment (j-f1) [inline packages/desktop-client/src/components/autocomplete/NewAutocomplete.tsx:18]: Why wouldn’t this be just `isNew`?

**Review by j-f1 — CHANGES_REQUESTED** [review]
One remaining issue is that if you click on the field and then press delete it will remove the typed in text but the placeholder will still be visible:
Screen Recording 2023-04-05 at 10 57 52

File: packages/desktop-client/src/components/autocomplete/NewAutocomplete.tsx (hunk 3 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/autocomplete/NewAutocomplete.tsx:18]: The `__isNew__` is set internally by `react-select` when you create a new autocomplete element. Not controlled by us actually.

**Comment by MatissJanis** [comment]
Fixed some more issues, but not the one you mentioned. Really not sure how we can solve that one.. but I'll continue cracking on it in a follow-up PR. Would you be OK with that?

**Review by j-f1 — APPROVED** [review]
Sure!

**Pull request merged**